### PR TITLE
Update d2bot status with waiting in line count

### PR DESCRIPTION
**Pull request by jaenster**
Actually wait for the queue in-line and make it visual @ the D2Bot#

### CODE REVIEW

File: d2bs/kolbot/D2BotLead.dbj [hunk 1]
```diff
@@ -392,10 +392,18 @@ MainSwitch:
...
-		D2Bot.updateStatus("Waiting...");
-		locationTimeout(StarterConfig.WaitInLineTimeout * 1e3, location);
-		ControlAction.click(6, 433, 433, 96, 32);
-
+		let positionInLine = ControlAction.getQueueTime();
+		if (positionInLine) {
+			D2Bot.updateStatus("Position in line " + positionInLine);
+
+			let nowLine;
+			while ((nowLine = ControlAction.getQueueTime())) {
```
Comment (emily785) [inline d2bs/kolbot/D2BotLead.dbj:400]: Tight loop, will burn cpu

**Comment by jaenster** [comment]
@emily785 thank you 💙

File: d2bs/kolbot/libs/OOG.js [hunk 2]
```diff
@@ -1016,6 +1016,23 @@ MainLoop:
 		return position;
 	},
 
+	getQueueTime: function() {
+		const control = getControl(4, 427, 234, 300, 100);
+
+		if (control && control.getText()) {
```
Comment (noah-) [inline d2bs/kolbot/libs/OOG.js:1022]: remove control.getText()

File: d2bs/kolbot/libs/OOG.js [hunk 3]
```diff
@@ -1016,6 +1016,23 @@ MainLoop:
...
+			const text = control.getText(); //You are in line to create a game.,Try joining a game to avoid waiting.,,Your position in line is: ÿc02912
+			if (text.indexOf(getLocaleString(11026)) !== -1) {
```
Comment (noah-) [inline d2bs/kolbot/libs/OOG.js:1024]: change to if (text && text.indexOf(...)

File: d2bs/kolbot/libs/OOG.js [hunk 4]
```diff
@@ -1016,6 +1016,23 @@ MainLoop:
...
+				const result = /ÿc0(\d*)/gm.exec(text);
+				if (result && typeof result[1] === 'string') {
+
```
Comment (noah-) [inline d2bs/kolbot/libs/OOG.js:1027]: random newline

Comment (jaenster) [reply d2bs/kolbot/libs/OOG.js:1027]: [eslint](https://gist.githubusercontent.com/Nishimura-Katsuo/2d6866666c7acf10047c486a15a7fe60/raw/99ef9c2995929c492ef856772ff346e0f19709cd/.eslintrc.js)

`	"padding-line-between-statements": ["error",
			{"blankLine": "always", "prev": "*", "next": ["return", "block-like"]},`

I dont like it either, but its your standard

File: d2bs/kolbot/D2BotLead.dbj [hunk 5]
```diff
@@ -392,10 +392,19 @@ MainSwitch:
 
 		break;
 	case 2: // Waiting In Line
-		D2Bot.updateStatus("Waiting...");
-		locationTimeout(StarterConfig.WaitInLineTimeout * 1e3, location);
-		ControlAction.click(6, 433, 433, 96, 32);
-
+		let positionInLine = ControlAction.getQueueTime();
```
Comment (noah-) [inline d2bs/kolbot/D2BotLead.dbj:395]: most of this is redundant
```
let line = 0;
let now;
while ((now = ControlAction.getQueueTime())) {
	if (line != now) {
		D2Bot.updateStatus("Position in line " + positionInLine);
		line = now;
	}
	delay(1000);
}
```

Comment (noah-) [reply d2bs/kolbot/D2BotLead.dbj:395]: Also consider keeping the "default" waiting in line delay, for example, if you are here and you read 0 for some reason (if that's possible) you may want some sort of delay to allow UI to catch up

File: d2bs/kolbot/libs/OOG.js (hunk 4 above)
Comment (noah-) [reply d2bs/kolbot/libs/OOG.js:1027]: I think nishi wrote the eslint, but nonetheless, feel free to submit change to the eslint as well

**Comment by jaenster** [comment]
yeah i prefer laz's version too, @laztheripper make pr already!

**Pull request closed without merging**